python_modules/dagster/dagster/_core/definitions/asset_selection.py

```diff
@@ -283,11 +283,35 @@ def without_checks(self) -> "AssetSelection":
 
     def __or__(self, other: "AssetSelection") -> "OrAssetSelection":
         check.inst_param(other, "other", AssetSelection)
-        return OrAssetSelection(self, other)
+
+        operands = []
```

Comment:
I don't know Python well for this case. But the only thing which changes between ``_and__`` and ``_or__``is the second if. Is it possible to put this repeated code into one function that takes the operator as an argument?

Reply:
Hi @SamirPS ! That's definitely possible in this case, although personally I think that might just add friction for a future person reading this code to understand what's happening in it.

That function would need to live as a standalone utility somewhere else in this file (so a future person would need to page around in this file to get the full context), and that function's purpose would be limited to these specific operators (it's unlikely that we'd come up with any other operators that need this sort of pattern in the future).